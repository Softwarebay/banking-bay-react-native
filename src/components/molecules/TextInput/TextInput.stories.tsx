import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { ViewStory } from 'utils/storybook/ViewStory';

import { TextInput } from './TextInput';

storiesOf('Molecules/TextInput', module)
	.addDecorator((getStory) => <ViewStory>{getStory()}</ViewStory>)
	.add('Default', () => (
		<TextInput placeholder={text('placeholder', 'Enter text')} />
	));
