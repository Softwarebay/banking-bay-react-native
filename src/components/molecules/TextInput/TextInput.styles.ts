import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	container: {
		height: 55,
		paddingHorizontal: 20,
		paddingVertical: 14,
	},
});
