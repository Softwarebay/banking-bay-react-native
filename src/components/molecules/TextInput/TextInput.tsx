import { forwardRef } from 'react';
import { StyleProp, TextStyle } from 'react-native';

import { Card, InputField, InputFieldHandle } from 'components/atoms';

import { styles } from './TextInput.styles';

export interface Props {
	placeholder: string;
	testID: string;
	style?: StyleProp<TextStyle>;
}

export const TextInput = forwardRef<InputFieldHandle, Props>(
	({ placeholder, style, testID }, ref) => {
		return (
			<Card style={[styles.container, style]} testID={testID}>
				<InputField
					ref={ref}
					keyboardType="default"
					placeholder={placeholder}
					secureTextEntry={false}
					style={styles.inputField}
					testID={`input-field-${testID}`}
				/>
			</Card>
		);
	}
);
