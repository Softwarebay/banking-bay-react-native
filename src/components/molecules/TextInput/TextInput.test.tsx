import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { TextInput } from './TextInput';

import type { Props } from './TextInput';

describe('TextInput', () => {
	const props: Props = {
		testID: 'text-input-test-id',
		placeholder: 'Placeholder',
		style: {
			height: 50,
		},
	};

	it('should match snapshot', () => {
		const tree = testRenderer.create(<TextInput {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render text input with default keyboard type', () => {
		render(<TextInput {...props} />);

		const inputField = screen.getByTestId('input-field-text-input-test-id');

		expect(inputField.props).toHaveProperty('keyboardType', 'default');
	});

	it('should display placeholder', () => {
		render(<TextInput {...props} />);

		const inputField = screen.getByTestId('input-field-text-input-test-id');

		expect(inputField.props).toHaveProperty('placeholder', 'Placeholder');
	});

	it('should apply additional styles', () => {
		render(<TextInput {...props} />);

		const input = screen.getByTestId('text-input-test-id');

		expect(input).toHaveStyle({
			height: 50,
		});
	});

	it('should handle type events', () => {
		render(<TextInput {...props} />);

		const inputField = screen.getByTestId('input-field-text-input-test-id');

		fireEvent.changeText(inputField, 'hello');

		expect(inputField.props).toHaveProperty('value', 'hello');
	});
});
