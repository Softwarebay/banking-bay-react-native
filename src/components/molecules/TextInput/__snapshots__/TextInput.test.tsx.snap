// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "row",
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "shadowColor": "#161966",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        Array [
          Object {
            "height": 55,
            "paddingHorizontal": 20,
            "paddingVertical": 14,
          },
          Object {
            "height": 50,
          },
        ],
      ],
    ]
  }
  testID="text-input-test-id"
>
  <TextInput
    autoCorrect={false}
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="Placeholder"
    secureTextEntry={false}
    style={
      Array [
        Object {
          "color": "#1B1D4D",
          "fontFamily": "MulishRegular",
          "fontSize": 15,
        },
        undefined,
      ]
    }
    testID="input-field-text-input-test-id"
    value=""
  />
</View>
`;
